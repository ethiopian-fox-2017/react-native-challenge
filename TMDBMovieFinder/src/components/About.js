import React from 'react'
import PropTypes from 'prop-types'
import {
  View,
  Text,
  Image,
  Button
} from 'react-native'

import Link from './Link'
import { styles } from '../styles'

export default class About extends React.Component {

  render () {
    const { navigation } = this.props
    return (
      <View style={styles.aboutBox}>
        <Image source={require('../img/tmdb1.png')} style={styles.logoTMDB} />
        <Text style={styles.aboutText}>A simple Movie Finder App powered by The Movie Database API to get the latest most popular movie</Text>
        <View>
          <Link url={'https://github.com/andrasat'}/>
        </View>
        <Button
          title="Go Back"
          color="#01d277"
          onPress={() => navigation.goBack()}
        />
      </View>
    )
  }
}

About.propTypes = {
  navigation: PropTypes.object.isRequired
}