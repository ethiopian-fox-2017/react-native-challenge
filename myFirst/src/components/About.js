import React, { Component } from 'react';
import { Image, Linking } from 'react-native';

import { Container, Content, Card, CardItem, Thumbnail, Text, Button, Icon, Left, Body, H1 } from 'native-base';


class About extends Component {
  static navigationOptions = {
    title: 'About',

  };
  render() {
    const { navigation } = this.props;
    return(
      <Container style={{ padding: 20 }}>

        <Content>
        <H1 style={{ textAlign: 'center' }}>About Me</H1>
          <Card style={{ flex: 0 }}>
            <CardItem>
              <Left>
                <Image
                  style={{width: 60, height: 60, borderRadius: 100}}
                  source={{uri: 'https://media.licdn.com/mpr/mpr/shrinknp_200_200/AAEAAQAAAAAAAAoLAAAAJDMwNWY5NTViLWVkNDItNGQ2YS1iZjVjLTIzODhhY2I3ZGJlOQ.jpg' }}
                />
                <Body>
                  <Text>Daily Information</Text>
                  <Text note>By: Anthony Juan</Text>
                </Body>
              </Left>
            </CardItem>
            <CardItem cardBody>
              <Image />
            </CardItem>
            <CardItem content>
              <Text>
                Aplikasi ini dibuat dengan tujuan untuk bersenang-senang, bila ada kesamaan nama pada karakter atau tokoh, itu adalah sebuah ketidak sengajaan
              </Text>
            </CardItem>
            <CardItem style={{justifyContent: 'space-around'}}>
            <Button className="buttonGit" transparent onPress={() => {Linking.openURL('https://github.com/anthonyjuan')}}>
              <Image
              style={{width: 30, height: 30, borderRadius: 100}}
              source={{uri:'http://www.techgirlz.org/wp-content/uploads/2016/12/github-logo.png'}}
              />
            </Button>

            </CardItem>
          </Card>
        </Content>
        <Button iconLeft light onPress={() => navigation.goBack()}>
           <Icon name='arrow-back' />
           <Text>Back</Text>
         </Button>
      </Container>
    )
  }
}

export default About;
