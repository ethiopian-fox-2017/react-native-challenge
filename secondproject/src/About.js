import React from 'react'

import {
  Text,
  View,
  Button,
} from 'react-native'

class About extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <View style={styles.container}>
        <Text>
          This is about
        </Text>
      </View>
    )
  }
}

const styles = {
  container: {
    flex: 1,
    backgroundColor: 'teal'
  }
}

export default About