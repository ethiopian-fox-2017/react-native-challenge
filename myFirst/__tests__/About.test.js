import 'react-native';
import React from 'react';
import About from '../src/components/About';
import { shallow } from 'enzyme';
import { Button } from 'native-base';
import { Image } from 'react-native';
import { Linking } from 'react-native';
import sinon from 'sinon';
// Note: test renderer must be required after react-native.

describe('testing about page', () => {
  test('renders correctly', () => {
    const tree = shallow(
      <About />
    );

    expect(tree).toMatchSnapshot();
  });

  test('image github open link when pressed', () => {
    const onPress = sinon.spy()
    const wrapper = shallow(<About />).find('.buttonGit')
    wrapper.setProps({onPress: onPress});
    wrapper.simulate('press');
    expect(onPress.calledOnce).toEqual(true);
  })
})
