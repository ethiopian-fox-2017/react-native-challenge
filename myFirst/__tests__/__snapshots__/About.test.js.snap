// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing about page renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Styled(Container)
    style={
        Object {
            "padding": 20,
          }
    }
    virtual={undefined}
>
    <Styled(Content)
        virtual={undefined}
    >
        <Styled(H1)
            style={
                Object {
                    "textAlign": "center",
                  }
            }
            virtual={undefined}
        >
            About Me
        </Styled(H1)>
        <Styled(Card)
            style={
                Object {
                    "flex": 0,
                  }
            }
            virtual={undefined}
        >
            <Styled(CardItem)
                virtual={undefined}
            >
                <Styled(Left)
                    virtual={undefined}
                >
                    <Image
                        source={
                            Object {
                                "uri": "https://media.licdn.com/mpr/mpr/shrinknp_200_200/AAEAAQAAAAAAAAoLAAAAJDMwNWY5NTViLWVkNDItNGQ2YS1iZjVjLTIzODhhY2I3ZGJlOQ.jpg",
                              }
                        }
                        style={
                            Object {
                                "borderRadius": 100,
                                "height": 60,
                                "width": 60,
                              }
                        }
                    />
                    <Styled(Body)
                        virtual={undefined}
                    >
                        <Styled(Text)
                            virtual={undefined}
                        >
                            Daily Information
                        </Styled(Text)>
                        <Styled(Text)
                            note={true}
                            virtual={undefined}
                        >
                            By: Anthony Juan
                        </Styled(Text)>
                    </Styled(Body)>
                </Styled(Left)>
            </Styled(CardItem)>
            <Styled(CardItem)
                cardBody={true}
                virtual={undefined}
            >
                <Image />
            </Styled(CardItem)>
            <Styled(CardItem)
                content={true}
                virtual={undefined}
            >
                <Styled(Text)
                    virtual={undefined}
                >
                    Aplikasi ini dibuat dengan tujuan untuk bersenang-senang, bila ada kesamaan nama pada karakter atau tokoh, itu adalah sebuah ketidak sengajaan
                </Styled(Text)>
            </Styled(CardItem)>
            <Styled(CardItem)
                style={
                    Object {
                        "justifyContent": "space-around",
                      }
                }
                virtual={undefined}
            >
                <Styled(Button)
                    className="buttonGit"
                    onPress={[Function]}
                    transparent={true}
                    virtual={undefined}
                >
                    <Image
                        source={
                            Object {
                                "uri": "http://www.techgirlz.org/wp-content/uploads/2016/12/github-logo.png",
                              }
                        }
                        style={
                            Object {
                                "borderRadius": 100,
                                "height": 30,
                                "width": 30,
                              }
                        }
                    />
                </Styled(Button)>
            </Styled(CardItem)>
        </Styled(Card)>
    </Styled(Content)>
    <Styled(Button)
        iconLeft={true}
        light={true}
        onPress={[Function]}
        virtual={undefined}
    >
        <Styled(Icon)
            name="arrow-back"
            virtual={undefined}
        />
        <Styled(Text)
            virtual={undefined}
        >
            Back
        </Styled(Text)>
    </Styled(Button)>
</Styled(Container)>,
  "nodes": Array [
    <Styled(Container)
      style={
            Object {
                  "padding": 20,
                }
      }
      virtual={undefined}
>
      <Styled(Content)
            virtual={undefined}
      >
            <Styled(H1)
                  style={
                        Object {
                              "textAlign": "center",
                            }
                  }
                  virtual={undefined}
            >
                  About Me
            </Styled(H1)>
            <Styled(Card)
                  style={
                        Object {
                              "flex": 0,
                            }
                  }
                  virtual={undefined}
            >
                  <Styled(CardItem)
                        virtual={undefined}
                  >
                        <Styled(Left)
                              virtual={undefined}
                        >
                              <Image
                                    source={
                                          Object {
                                                "uri": "https://media.licdn.com/mpr/mpr/shrinknp_200_200/AAEAAQAAAAAAAAoLAAAAJDMwNWY5NTViLWVkNDItNGQ2YS1iZjVjLTIzODhhY2I3ZGJlOQ.jpg",
                                              }
                                    }
                                    style={
                                          Object {
                                                "borderRadius": 100,
                                                "height": 60,
                                                "width": 60,
                                              }
                                    }
                              />
                              <Styled(Body)
                                    virtual={undefined}
                              >
                                    <Styled(Text)
                                          virtual={undefined}
                                    >
                                          Daily Information
                                    </Styled(Text)>
                                    <Styled(Text)
                                          note={true}
                                          virtual={undefined}
                                    >
                                          By: Anthony Juan
                                    </Styled(Text)>
                              </Styled(Body)>
                        </Styled(Left)>
                  </Styled(CardItem)>
                  <Styled(CardItem)
                        cardBody={true}
                        virtual={undefined}
                  >
                        <Image />
                  </Styled(CardItem)>
                  <Styled(CardItem)
                        content={true}
                        virtual={undefined}
                  >
                        <Styled(Text)
                              virtual={undefined}
                        >
                              Aplikasi ini dibuat dengan tujuan untuk bersenang-senang, bila ada kesamaan nama pada karakter atau tokoh, itu adalah sebuah ketidak sengajaan
                        </Styled(Text)>
                  </Styled(CardItem)>
                  <Styled(CardItem)
                        style={
                              Object {
                                    "justifyContent": "space-around",
                                  }
                        }
                        virtual={undefined}
                  >
                        <Styled(Button)
                              className="buttonGit"
                              onPress={[Function]}
                              transparent={true}
                              virtual={undefined}
                        >
                              <Image
                                    source={
                                          Object {
                                                "uri": "http://www.techgirlz.org/wp-content/uploads/2016/12/github-logo.png",
                                              }
                                    }
                                    style={
                                          Object {
                                                "borderRadius": 100,
                                                "height": 30,
                                                "width": 30,
                                              }
                                    }
                              />
                        </Styled(Button)>
                  </Styled(CardItem)>
            </Styled(Card)>
      </Styled(Content)>
      <Styled(Button)
            iconLeft={true}
            light={true}
            onPress={[Function]}
            virtual={undefined}
      >
            <Styled(Icon)
                  name="arrow-back"
                  virtual={undefined}
            />
            <Styled(Text)
                  virtual={undefined}
            >
                  Back
            </Styled(Text)>
      </Styled(Button)>
</Styled(Container)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <About />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": About {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Styled(Container)
          style={
                    Object {
                              "padding": 20,
                            }
          }
          virtual={undefined}
>
          <Styled(Content)
                    virtual={undefined}
          >
                    <Styled(H1)
                              style={
                                        Object {
                                                  "textAlign": "center",
                                                }
                              }
                              virtual={undefined}
                    >
                              About Me
                    </Styled(H1)>
                    <Styled(Card)
                              style={
                                        Object {
                                                  "flex": 0,
                                                }
                              }
                              virtual={undefined}
                    >
                              <Styled(CardItem)
                                        virtual={undefined}
                              >
                                        <Styled(Left)
                                                  virtual={undefined}
                                        >
                                                  <Image
                                                            source={
                                                                      Object {
                                                                                "uri": "https://media.licdn.com/mpr/mpr/shrinknp_200_200/AAEAAQAAAAAAAAoLAAAAJDMwNWY5NTViLWVkNDItNGQ2YS1iZjVjLTIzODhhY2I3ZGJlOQ.jpg",
                                                                              }
                                                            }
                                                            style={
                                                                      Object {
                                                                                "borderRadius": 100,
                                                                                "height": 60,
                                                                                "width": 60,
                                                                              }
                                                            }
                                                  />
                                                  <Styled(Body)
                                                            virtual={undefined}
                                                  >
                                                            <Styled(Text)
                                                                      virtual={undefined}
                                                            >
                                                                      Daily Information
                                                            </Styled(Text)>
                                                            <Styled(Text)
                                                                      note={true}
                                                                      virtual={undefined}
                                                            >
                                                                      By: Anthony Juan
                                                            </Styled(Text)>
                                                  </Styled(Body)>
                                        </Styled(Left)>
                              </Styled(CardItem)>
                              <Styled(CardItem)
                                        cardBody={true}
                                        virtual={undefined}
                              >
                                        <Image />
                              </Styled(CardItem)>
                              <Styled(CardItem)
                                        content={true}
                                        virtual={undefined}
                              >
                                        <Styled(Text)
                                                  virtual={undefined}
                                        >
                                                  Aplikasi ini dibuat dengan tujuan untuk bersenang-senang, bila ada kesamaan nama pada karakter atau tokoh, itu adalah sebuah ketidak sengajaan
                                        </Styled(Text)>
                              </Styled(CardItem)>
                              <Styled(CardItem)
                                        style={
                                                  Object {
                                                            "justifyContent": "space-around",
                                                          }
                                        }
                                        virtual={undefined}
                              >
                                        <Styled(Button)
                                                  className="buttonGit"
                                                  onPress={[Function]}
                                                  transparent={true}
                                                  virtual={undefined}
                                        >
                                                  <Image
                                                            source={
                                                                      Object {
                                                                                "uri": "http://www.techgirlz.org/wp-content/uploads/2016/12/github-logo.png",
                                                                              }
                                                            }
                                                            style={
                                                                      Object {
                                                                                "borderRadius": 100,
                                                                                "height": 30,
                                                                                "width": 30,
                                                                              }
                                                            }
                                                  />
                                        </Styled(Button)>
                              </Styled(CardItem)>
                    </Styled(Card)>
          </Styled(Content)>
          <Styled(Button)
                    iconLeft={true}
                    light={true}
                    onPress={[Function]}
                    virtual={undefined}
          >
                    <Styled(Icon)
                              name="arrow-back"
                              virtual={undefined}
                    />
                    <Styled(Text)
                              virtual={undefined}
                    >
                              Back
                    </Styled(Text)>
          </Styled(Button)>
</Styled(Container)>,
        "_debugID": 2,
        "_renderedOutput": <Styled(Container)
          style={
                    Object {
                              "padding": 20,
                            }
          }
          virtual={undefined}
>
          <Styled(Content)
                    virtual={undefined}
          >
                    <Styled(H1)
                              style={
                                        Object {
                                                  "textAlign": "center",
                                                }
                              }
                              virtual={undefined}
                    >
                              About Me
                    </Styled(H1)>
                    <Styled(Card)
                              style={
                                        Object {
                                                  "flex": 0,
                                                }
                              }
                              virtual={undefined}
                    >
                              <Styled(CardItem)
                                        virtual={undefined}
                              >
                                        <Styled(Left)
                                                  virtual={undefined}
                                        >
                                                  <Image
                                                            source={
                                                                      Object {
                                                                                "uri": "https://media.licdn.com/mpr/mpr/shrinknp_200_200/AAEAAQAAAAAAAAoLAAAAJDMwNWY5NTViLWVkNDItNGQ2YS1iZjVjLTIzODhhY2I3ZGJlOQ.jpg",
                                                                              }
                                                            }
                                                            style={
                                                                      Object {
                                                                                "borderRadius": 100,
                                                                                "height": 60,
                                                                                "width": 60,
                                                                              }
                                                            }
                                                  />
                                                  <Styled(Body)
                                                            virtual={undefined}
                                                  >
                                                            <Styled(Text)
                                                                      virtual={undefined}
                                                            >
                                                                      Daily Information
                                                            </Styled(Text)>
                                                            <Styled(Text)
                                                                      note={true}
                                                                      virtual={undefined}
                                                            >
                                                                      By: Anthony Juan
                                                            </Styled(Text)>
                                                  </Styled(Body)>
                                        </Styled(Left)>
                              </Styled(CardItem)>
                              <Styled(CardItem)
                                        cardBody={true}
                                        virtual={undefined}
                              >
                                        <Image />
                              </Styled(CardItem)>
                              <Styled(CardItem)
                                        content={true}
                                        virtual={undefined}
                              >
                                        <Styled(Text)
                                                  virtual={undefined}
                                        >
                                                  Aplikasi ini dibuat dengan tujuan untuk bersenang-senang, bila ada kesamaan nama pada karakter atau tokoh, itu adalah sebuah ketidak sengajaan
                                        </Styled(Text)>
                              </Styled(CardItem)>
                              <Styled(CardItem)
                                        style={
                                                  Object {
                                                            "justifyContent": "space-around",
                                                          }
                                        }
                                        virtual={undefined}
                              >
                                        <Styled(Button)
                                                  className="buttonGit"
                                                  onPress={[Function]}
                                                  transparent={true}
                                                  virtual={undefined}
                                        >
                                                  <Image
                                                            source={
                                                                      Object {
                                                                                "uri": "http://www.techgirlz.org/wp-content/uploads/2016/12/github-logo.png",
                                                                              }
                                                            }
                                                            style={
                                                                      Object {
                                                                                "borderRadius": 100,
                                                                                "height": 30,
                                                                                "width": 30,
                                                                              }
                                                            }
                                                  />
                                        </Styled(Button)>
                              </Styled(CardItem)>
                    </Styled(Card)>
          </Styled(Content)>
          <Styled(Button)
                    iconLeft={true}
                    light={true}
                    onPress={[Function]}
                    virtual={undefined}
          >
                    <Styled(Icon)
                              name="arrow-back"
                              virtual={undefined}
                    />
                    <Styled(Text)
                              virtual={undefined}
                    >
                              Back
                    </Styled(Text)>
          </Styled(Button)>
</Styled(Container)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <About />,
}
`;
